main.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"github.com/cloudogu/k8s-prometheus/auth/configuration"
+	"github.com/cloudogu/k8s-prometheus/auth/prometheus"
+	"github.com/cloudogu/k8s-prometheus/auth/proxy"
+	"github.com/cloudogu/k8s-prometheus/auth/serviceaccount"
+	"github.com/gin-gonic/gin"
+	"log/slog"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+func main() {
+	config, err := configuration.ReadConfigFromEnv()
+	if err != nil {
+		//panic(err)
+	}
+
+	gin.SetMode(gin.ReleaseMode)
+
+	configureLogger(config)
+
+	manager := prometheus.NewManager(config.WebConfigFile)
+
+	serviceAccountSrv := serviceaccount.CreateServer(config, manager)
+	go func() {
+		slog.Info("service-account-provider started", "addr", serviceAccountSrv.Addr)
+		if err := serviceAccountSrv.ListenAndServe(); err != nil && !errors.Is(err, http.ErrServerClosed) {
+			slog.Error("error starting service-account server.", "err", err)
+		}
+	}()
+
+	proxySrv := proxy.CreateServer(config, manager)
+	go func() {
+		slog.Info("auth-proxy started", "addr", proxySrv.Addr)
+		if err := proxySrv.ListenAndServe(); err != nil && !errors.Is(err, http.ErrServerClosed) {
+			slog.Error("error starting auth-proxy server.", "err", err)
+		}
+	}()
+
+	// Wait for interrupt signal to gracefully shut down the server with a timeout of 5 seconds.
+	quit := make(chan os.Signal)
+	signal.Notify(quit, syscall.SIGINT, syscall.SIGTERM)
```

Comment:
🚫 **[checkstyle]** <govet> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>sigchanyzer: misuse of unbuffered os.Signal channel as argument to signal.Notify